src/debugAdapter/goDebug.ts

```diff
@@ -1587,6 +1585,75 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	private logDelveError(err: any, message: string) {
+		if (err) {
+			let errorMessage = err.toString();
+			// Handle unpropagated fatalpanic errors with a more user friendly message:
+			// https://github.com/microsoft/vscode-go/issues/1903#issuecomment-460126884
+			// https://github.com/go-delve/delve/issues/852
+			// This affects macOS only although we're agnostic of the OS at this stage, only handle the error
+			if (errorMessage === 'bad access') {
+				errorMessage = 'unpropagated fatalpanic: signal SIGSEGV (EXC_BAD_ACCESS). This fatalpanic is not traceable on macOS, see https://github.com/go-delve/delve/issues/852';
+			}
+
+			logError(message + ' - ' + errorMessage);
+
+			if (err.toString() === 'bad access') {
+				logError('WARNING: this stack might not be from the expected active goroutine');
```

Comment:
I think this might suffice for now, what do you think @jhendrixMSFT?

Reply:
I would reword this to "...the faulting goroutine".
I also wonder if it would just be better to omit the stack trace entirely given that it might not be accurate (we could add it later if there's sufficient feedback).  @ramya-rao-a do you have an opinion about this?